float xc
float yc
float zc
float a1
float a2
float d3
â€”
float theta1
float theta2
float d3out